echopype/echodata/echodata.py

```diff
@@ -518,11 +518,21 @@ def _clip_by_time_dim(external_ds, ext_time_dim_name):
         ]
         for platform_var in scalar_vars:
             ext_var = mappings_expanded[platform_var]["external_var"]
-            # Replace the scalar value and add a history attribute
-            platform[platform_var].data = float(extra_platform_data[ext_var].data)
-            platform[platform_var] = platform[platform_var].assign_attrs(
-                **{"history": f"{history_attr}. From external {ext_var} variable."}
-            )
+            if platform_var.startswith("lon") or platform_var.startswith("lat"):
+                platform[platform_var].data = np.array([extra_platform_data[ext_var].data])
+                platform[platform_var] = platform[platform_var].assign_coords(
+                    **{
+                        platform[platform_var].dims[0]: [
+                            self["Sonar/Beam_group1"]["ping_time"].data[0]
+                        ]
+                    }
+                )
```

Comment:
Could you add a comment here to say that these handlings are specific to lat/lon variables only, and be specific that we are using the first ping_time value as the timestamp?

Reply:
Updated, added a comment to specify the use case!